buffer/src/main/java/io/netty/buffer/api/adaptor/ByteBufAdaptor.java

```diff
@@ -128,6 +128,22 @@ public static Buffer extractOrCopy(BufferAllocator allocator, ByteBuf byteBuf) {
         }
     }
 
+    /**
+     * Convert the given {@link Buffer} into a {@link ByteBuf} using the most optimal method.
+     * The contents of both buffers will be mirrored.
+     *
+     * @param buffer The {@link Buffer} to convert into a {@link ByteBuf}.
+     * @return A {@link ByteBuf} that uses the given {@link Buffer} as a backing store.
+     */
+    public static ByteBuf intoByteBuf(Buffer buffer) {
+        if (buffer instanceof ByteBufConvertible) {
+            ByteBufConvertible convertible = (ByteBufConvertible) buffer;
+            return convertible.asByteBuf();
+        } else {
+            return new ByteBufAdaptor(new ByteBufAllocatorAdaptor(), buffer, buffer.capacity());
```

Comment:
Should we reuse the `ByteBufAllocatorAdaptor`? perhaps use the global allocator?

Reply:
Good point. Fixed in https://github.com/netty/netty/pull/11720